File: examples/python/rabbitmq-producer-consumer/consumer.py
import os
import pika

url = os.getenv("RABBITMQ_URL", "amqp://admin:change_me@localhost:5672/app")
params = pika.URLParameters(url)
conn = pika.BlockingConnection(params)
ch = conn.channel()

ch.queue_declare(queue="app.queue", durable=True)
ch.queue_bind(queue="app.queue", exchange="app.exchange", routing_key="app.created")

print("[consumer] waiting...")
for method, props, body in ch.consume("app.queue", inactivity_timeout=120):
    if body:
        print("[consumer] got", body.decode())
        ch.basic_ack(method.delivery_tag)
        break
conn.close()

File: examples/nodejs/rabbitmq-producer-consumer/consumer.js
import amqp from 'amqplib';

const url = process.env.RABBITMQ_URL || 'amqp://admin:change_me@localhost:5672/app';
const conn = await amqp.connect(url);
const ch = await conn.createChannel();

await ch.assertQueue('app.queue', { durable: true });
await ch.bindQueue('app.queue', 'app.exchange', 'app.created');
console.log('[consumer] waiting ...');

ch.consume('app.queue', (msg) => {
  if (!msg) return;
  console.log('[consumer] got', msg.content.toString());
  ch.ack(msg);
});
